spinalcordtoolbox/reports/qc.py

```diff
@@ -585,7 +585,7 @@ def update_description_file(self):
         logger.debug('Description file: %s', self.qc_results)
 
         # Use a mutex on a hash of the QC path, so that we use a unique mutex per target QC report
-        with mutex(f"sct_qc-{os.path.basename(path_qc)}-{md5(self.path_qc.encode('utf-8')).hexdigest()}"):
+        with mutex(f"sct_qc-{md5(os.path.realpath(path_qc).encode('utf-8')).hexdigest()}"):
```

Comment:
I think the reason I had added `basename(path_qc)` outside of the hash function was not necessarily for uniqueness' sake, but for debugging purposes (so that we could have a plaintext way of identifying which lock file was associated with which QC folder).

I've seen other names for QC folders besides `qc` e.g. `qcSingleSubj` (for the SCT course), `qc_example_data` (for `batch_processing.sh`), `test`, etc..

Reply:
Ok! That makes sense, I can add it back. But I should extract the basename _after_ taking `realpath`, so that different symlinks to the same qc folder still get the same lock name.